src/app/plot.py

```diff
@@ -0,0 +1,221 @@
+
+import numpy as np
+import pandas as pd
+import altair as alt
+import geopandas as gpd
+
+
+data = pd.read_csv("./../../data/processed/cleaned_salaries.csv")
+
+def plot_22(xmax, con):
+    
+    source = data[(data["Age"]>0) & (data["Salary_USD"]<=xmax[1])]
+    if con == "World" or con is None:
+        pass
+    else:
+        source = source[source["Country"] == con]
+
+    
+    alt.themes.enable("default")
+
+    chart = alt.Chart(source).mark_rect().encode(
+        x = alt.X("Age:Q", bin=alt.Bin(maxbins=60), title=None),
+        y = alt.Y("Salary_USD:Q", bin=alt.Bin(maxbins=40), 
+                  title="Salary in USD", axis=alt.Axis(format='~s')),
+        tooltip='count()',
+        color=alt.Color('count()',scale=alt.Scale(scheme='greenblue'), legend=alt.Legend(title='Tot. Records')),
+    ).properties(
+        title = "Heatmap of selected country",
+        width=335,
+        height=270,
+    )
+    
+    bar = alt.Chart(source).mark_bar().encode(
+        x='Age:Q',
+        y='count()',
+    ).properties(
+        width=335,
+        height=170,
+    )
+    
+    fchart = alt.vconcat(chart, bar, spacing=0)
+    
+    return fchart.to_html()
+
+
+def plot_12(xcon):
+    
+    genders = ['Male', 'Female', 'A different identity']
+    if xcon == "World" or xcon is None:
+        source = data[data["GenderSelect"].isin(genders)]
+    else:
+        source = data[(data["Country"] == xcon) & (data["GenderSelect"].isin(genders))]
+        
+    chart = alt.Chart(source).mark_boxplot().encode(
+        x=alt.X("Salary_USD:Q", 
+                title="Salary in USD", 
+                axis=alt.Axis(format='~s'),
+                scale=alt.Scale(zero=False)),
+        y=alt.Y('GenderSelect', title="Gender"),
+        tooltip='count()',
+        color=alt.Color('GenderSelect', title='Gender')
+    ).configure_legend(
+        orient='bottom'
+    ).properties(
+        title='Boxplot by gender',
+        projection={"type":'mercator'},
+        width=575,
+        height=180
+    ).interactive()
+    
+    return chart.to_html()
+
+
+def plot_21(value):
+    
+    education_order = ["Less than bachelor's degree", "Bachelor's degree", 
+                       "Master's degree", "Doctoral degree"]
+    
+    if value == "World" or value is None:
+        country = data
+    else:
+        country = data.query("Country == @value")
+        
+    for idx, i in enumerate(country["FormalEducation"]):
+        if i in education_order[1:]:
+            continue
+        else:
+            print("Change")
```

Comment:
Could we please delete this print statement?

Reply:
deleted